scripts/generate-releases.sh

```diff
@@ -2,64 +2,89 @@
 
 set -euo pipefail
 
-if [[ "$#" -lt 1 || "$#" -gt 3 ]]; then
-    echo "USAGE: ./generate-releases.sh <ENVIRONMENT> [COMMIT] [BRANCH]" >&2
-    echo "" >&2
-    echo "ENVIRONMENT - allowed values: staging|prod" >&2
-    echo "COMMIT - a 40 character-long SHA of the commit to pull Snapshots only with this commit label for the Release. Default: currently checked out commit" >&2
-    echo "BRANCH - an optional parameter to specify git branch name for filtering snapshots by having branch name in annotations. Default: currently checked out branch" >&2
-    echo "" >&2
-    echo "You must have your KUBECONFIG point to the Konflux cluster, see https://spaces.redhat.com/pages/viewpage.action?pageId=407312060#HowtoeverythingKonfluxforRHACS-GettingocCLItoworkwithKonflux." >&2
-    exit 1
-fi
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd)"
+source "$SCRIPT_DIR/helpers.sh"
 
-ENVIRONMENT="$1"
-COMMIT="${2:-$(git rev-parse HEAD)}"
-BRANCH="${3:-$(git rev-parse --abbrev-ref HEAD)}"
+# Validate the input parameters and ensure they are correct.
+validate_parameters() {
+    if [[ "$#" -lt 1 || "$#" -gt 3 ]]; then
+        echo "USAGE: ./generate-releases.sh <ENVIRONMENT> [COMMIT] [BRANCH]" >&2
+        echo "" >&2
+        echo "ENVIRONMENT - allowed values: staging|prod" >&2
+        echo "COMMIT - a SHA of the commit to pull Snapshots only with this commit label for the Release." >&2
+        echo "If provided commit SHA is less than 40 characters then it will be expanded to a full 40-characters long SHA. Default: currently checked out commit" >&2
+        echo "BRANCH - an optional parameter to specify git branch name for filtering snapshots by having branch name in annotations. Default: currently checked out branch" >&2
+        echo "" >&2
+        echo "You must have your KUBECONFIG point to the Konflux cluster, see https://spaces.redhat.com/pages/viewpage.action?pageId=407312060#HowtoeverythingKonfluxforRHACS-GettingocCLItoworkwithKonflux." >&2
+        exit 1
+    fi
+
+    environment="$1"
+    commit="${2:-$(git rev-parse HEAD)}"
+    branch="${3:-$(git rev-parse --abbrev-ref HEAD)}"
+
+    commit=$(expand_commit "$commit")
+    validate_branch "$branch"
```

Comment:
[nit] This would benefit from re-ordering
```suggestion
    commit="${2:-$(git rev-parse HEAD)}"
    commit="$(expand_commit "$commit")"

    branch="${3:-$(git rev-parse --abbrev-ref HEAD)}"
    validate_branch "$branch"
```